CDTDatastore/touchdb/TDPuller.m

```diff
@@ -398,15 +460,83 @@ - (void)pullRemoteRevision:(TD_Revision*)rev
     [dl start];
 }
 
+- (void)pullBulkRevisionsBulkGet:(NSArray*)bulkRevs
+{
+    // http://wiki.apache.org/couchdb/HTTP_Bulk_Document_API
+    NSUInteger nRevs = bulkRevs.count;
+    if (nRevs == 0) return;
+    CDTLogInfo(CDTREPLICATION_LOG_CONTEXT, @"%@ bulk-fetching (via _bulk_get) %u remote revisions...", self,
+               (unsigned)nRevs);
+    CDTLogVerbose(CDTREPLICATION_LOG_CONTEXT, @"%@ bulk-fetching (via _bulk_get) remote revisions: %@", self, bulkRevs);
+    
+    [self asyncTaskStarted];
+    ++_httpConnectionCount;
+    
+    // body needs to be in form:
+    // {"docs":[{"id":"1-foo","rev":"rev123","atts_since":["1-foo,...]}]}
+    NSArray* keys = [bulkRevs my_map:^(TD_Revision* rev) {
+        NSArray* knownRevs = [_db getPossibleAncestorRevisionIDs:rev limit:kMaxNumberOfAttsSince];
+        return @{@"id": rev.docID,
+                 @"rev": rev.revID,
+                 @"atts_since": knownRevs != nil ? knownRevs : @[]};
+    }];
+    
+    NSDictionary *requestBody = @{@"docs": keys};    
+    NSMutableArray* remainingRevs = [bulkRevs mutableCopy];
+    __weak TDPuller* weakSelf = self;
+
+    [self sendAsyncRequest:@"POST"
+                      path:@"_bulk_get?revs=true&attachments=true"
+                      body:requestBody
+              onCompletion:^(id result, NSError* error) {
+                  __strong TDPuller* strongSelf = weakSelf;
+                  if (error) {
+                      strongSelf.error = error;
+                      [strongSelf revisionFailed];
+                      strongSelf.changesProcessed+=bulkRevs.count;
+                  } else {
+                      // unpack revisions and queue them in _downloadsToInsert
+                      NSArray *results = $castIf(NSArray, result[@"results"]);
```

Comment:
ok, so the `result` object that is passed in to the completion is always going to be an `NSDictionary` or `nil`?